README.md

```diff
@@ -1 +1,102 @@
-# hyperformula-demos
\ No newline at end of file
+# hyperformula-demos
+
+This repository contains demos and usage examples for the [HyperFormula](https://github.com/handsontable/hyperformula) library. Some of them are embedded in the [HyperFormula documentation](https://handsontable.github.io/hyperformula/).
+
+## Branches
+
+We keep:
+- **develop** branch
+- version branches e.g. **1.0.x**, **1.1.x**, **2.0.x**
+
+The demos on each version branch work with the corresponding version of HyperFormula.
+
+## Running the demos
+
+Each directory contains one demo. 
+
+*Make sure to use the right version of the demos. E.g. for **hyperformula@1.3.1** you should use the code from branch **1.3.x**.*
+
+### In CodeSandbox.io
+
+The demos are embedded in HyperFormula documentation using CodeSandbox.io so this should be treated as the primary execution environment for the demos.
+
+CodeSandbox.io uses `node@16.12.0` to run the demos as [configured](https://codesandbox.io/docs/configuration#sandbox-configuration) in `sandbox.config.json` file.
+
+The CodeSandbox.io links in the documentation have this form:
+```js
+`https://codesandbox.io/embed/github/handsontable/hyperformula-demos/tree/${branchName}/${directory}?autoresize=1&fontsize=11&hidenavigation=1&theme=light&view=preview`
+```
+
+E.g. a link for demo **basic-usage** from branch **2.0.x** looks like this:
+https://codesandbox.io/embed/github/handsontable/hyperformula-demos/tree/2.0.x/basic-usage?autoresize=1&fontsize=11&hidenavigation=1&theme=light&view=preview
+
+
+### Locally
+
+The demos use different frameworks and technologies but all of them can be run using `npm run start` command.
+
+*We recommend using Node v16 to run the demos.*
+
+```bash
+# in hyperformula-demos directory
+cd basic-usage # or any other demo
+npm install
+npm run start
+```
+
+For most demos the browser opens automatically with the correct URL. If it didn't, copy the URL from the console output and paste it into the browser.
+
+Every demo prints the HyperFormula version to the console.
+
+## Development & maintenance
+
+### Branching policy
+
+On the **develop** branch we:
+- create new demos
+- improve the existing demos
+- adjust the demos to the recent changes in HyperFormula's API
+
+For each *major* and *minor* release of HyperFormula a new branch is created from **develop** with name indicating the released version.
+
+The bugfixes for the previous versions are applied to the appropriate branches.
+
+### Updating HyperFormula version for all demos
+
+This command updates the `package.json` file in all demos to use version *x.y.z* of hyperformula package:
+
+```bash
+# in hyperformula-demos directory
+sh set-hyperformula-version.sh x.y.z
```

Comment:
@sequba 
When running this command (on MacOS 10.15) I get this error:

```
sed: 1: "./react-demo/package.json": invalid command code .
```

Not sure if I'm doing something wrong?

Reply:
It turns out MacOS implementation of `sed` is non-standard. I changed the script, so it should work for both GNU and MacOS versions of `sed`. @jansiegel Can you confirm that?

Reply:
This seems to work:

```
find . -maxdepth 2 -name package.json | xargs sed -i '' 's/"hyperformula": "[^"]*"/"hyperformula": "'$1'"/g'
```

Reply:
@warpech your version doesn't work with GNU sed. I commited another version:
```
find . -maxdepth 2 -name package.json -print0 | xargs -0 sed -i'' 's/"hyperformula": "[^"]*"/"hyperformula": "'$1'"/g'
```
@warpech, @jansiegel Can you try this one? I don't have access to MacOS PC to test it myself.

Reply:
Unfortunately, this gives me an error:

```
sed: 1: "./react-demo/package.json": invalid command code .
```

The problem is in the lack of space after `-i`. The following command works:

```
find . -maxdepth 2 -name package.json -print0 | xargs -0 sed -i '' 's/"hyperformula": "[^"]*"/"hyperformula": "'$1'"/g'
```

Reply:
After [03f00aa](https://github.com/handsontable/hyperformula-demos/pull/7/commits/03f00aab0669969c2a085703608e1184be98d502), script `set-hyperformula-version.sh` works ok on my macOS. Thanks!